Paymetheus/ViewModels/PurchaseTicketsViewModel.cs

```diff
@@ -0,0 +1,404 @@
+        public double PoolFees
+        {
+            get { return _poolFees; }
+            set
+            {
+                var _testPoolFees = value * 100.0;
+                if (_testPoolFees != Math.Floor(_testPoolFees))
+                {
+                    _poolFees = 0.0;
+                    throw new ArgumentException("Bad pool fees; pool fees must be between 0.00 and 100.00%");
+                }
+
+                if (_testPoolFees > 10000.0)
+                {
+                    _poolFees = 0.0;
+                    throw new ArgumentException("Bad pool fees; pool fees must be less than 100.00%");
+                }
+
+                if (_testPoolFees < 1.0)
+                {
+                    _poolFees = 0.0;
+                    throw new ArgumentException("Bad pool fees; pool fees must be greater than 0.00%");
+                }
+
+                _poolFees = value;
+                _settingsAreValid();
+            }
+        }
+
+        private long _number = 0;
+        public long Number
+        {
+            get { return _number; }
+            set
+            {
+                if (value < 0)
+                {
+                    _number = 0;
+                    throw new ArgumentException("Negative number of tickets passed");
+                }
+
+                if (_selectedAccount == null)
+                {
+                    _number = 0;
+                    throw new ArgumentException("No account selected");
+                }
+
+                if (_selectedAccount.Balances.SpendableBalance <= 0) {
+                    _number = 0;
+                    throw new ArgumentException("Empty account");
+                }
+
+                if ((Amount)(_stakeDifficultyProperties.Price * value) > _selectedAccount.Balances.SpendableBalance)
+                {
+                    _number = 0;
+                    string errorString = "Not enough funds; have " +
+                        _selectedAccount.Balances.SpendableBalance.ToString() + " want " +
+                        ((Amount)(_stakeDifficultyProperties.Price * value)).ToString();
+                    throw new ArgumentException(errorString);
+                }
+
+                _number = value;
+                _settingsAreValid();
+            }
+        }
+
+        // The default expiry is 16.
+        private int _expiry = 16;
+        private int _minExpiry = 2;
+        public int Expiry
+        {
+            get { return _expiry; }
+            set
+            {
+                if (value <= _minExpiry)
+                {
+                    _expiry = 0;
+                    throw new ArgumentException("Expiry must be a minimum of 2 blocks");
+                }
+
+                _expiry = value;
+                _settingsAreValid();
+            }
+        }
+
+        // TODO Declare this as a global somewhere?
+        private Amount _minFeeAmount = Denomination.Decred.AmountFromDouble(0.01000000);
```

Comment:
hahaah no. we are not using doubles in our code.

```
private const long MinFeeAmount = (long)1e6;
private const long MaxFeeAmount = (long)1e8 - 1;
```

Unfortunately we can't declare const Amounts but since Amounts and longs can be implicitly cast to each other, this will work nicely.